### ISSUE TITLE
Fix request: false positive AvoidConcatInLoop

### ISSUE THREAD

**Issue by stokpop** [jborgers/PMD-jPinpoint-rules#564]
The concat is not part of the loop. Added most simplified case, but also one with extra `switch` statement which was also in original code.

```java
import java.util.List;

class Foo {

    private static final String PACKAGE_PREFIX = "PREFIX";

    public void doSomethingSimple(String name, List<String> options) {
        for (String option : options) {
            String loggerName = name;
            if (option.startsWith("j")) {
                loggerName = PACKAGE_PREFIX + name; // <-- false positive
            }
            setLevels(loggerName);
        }
    }

    public void doSomethingSwitch(String name, List<String> options) {
        for (String option : options) {
            switch(option) {
                case "j":
                    String loggerName = name;
                    if (option.startsWith("j")) {
                        loggerName = PACKAGE_PREFIX + name; // <-- false positive
                    }
                    setLevels(loggerName);
                    break;
                default:
                    setLevels(name);
            }
        }
    }

    private void setLevels(String loggerName) {
        // do something
    }
}
```
